lib/improver/utilities/temporal.py

```diff
@@ -627,10 +626,20 @@ def process(self, cube_t0, cube_t1):
         time_list = self.construct_time_list(initial_time, final_time)
         cubes = iris.cube.CubeList([cube_t0, cube_t1])
         cube = merge_cubes(cubes)
+
         interpolated_cube = cube.interpolate(time_list, iris.analysis.Linear())
+
+        # iris.analysis.Linear() modifies the dtype of time and forecast_period
+        # coords so need to revert back
+        dtype_time = cube_t0.coord('time').points.dtype
+        dtype_fp = cube_t0.coord('forecast_period').points.dtype
+
         interpolated_cubes = iris.cube.CubeList()
         for single_time in interpolated_cube.slices_over('time'):
-            check_cube_not_float64(single_time, fix=True)
+            coord_time = single_time.coord('time')
```

Comment:
Do any unit tests need updating after this change?

Reply:
No because the unit tests are not making use of set_up_cube_variable at the moment. I was going to update these when I add the extra functionality to the interpolation code in the next PR.

Reply:
I would also like to see a new test that coordinate datatype is unchanged.  But since the other work is happening in parallel, it can probably wait for that next PR.